src/sage/groups/perm_gps/partn_ref2/refinement_generic.pyx

```diff
@@ -941,3 +940,40 @@ cdef class PartitionRefinement_generic:
         """
         if BACKTRACK_WITHLATEX_DEBUG:
             self._latex_debug_string += "]\n"
+
+
+cdef int PS_first_smallest_PR(PartitionStack *PS, bitset_t b, int *second_pos=NULL,
+                              PartitionRefinement_generic partn_ref_alg=None):
+    """
+    Find the first occurrence of the smallest cell of size greater than one,
+    which is admissible (checked by the function ``test_allowance``).
+    Its entries are stored to b and its minimum element is returned.
+
+    This generalizes :func:`sage.groups.perm_gps.partn_ref.data_structures.PS_first_smallest`.
+    """
+    cdef int i = 0, j = 0, location = 0, n = PS.degree
+    bitset_zero(b)
+    while 1:
```

Comment:
`while True` might be better. 
Furthermore, this loop has no control on the max value of `I `. Isn't it safer to use a `for` loop combined with a `else` statement to raise an error in case `break` is never called ?